### ISSUE TITLE
Optimistic Unlocking

### ISSUE THREAD

**Issue by smombartz** [unlock-protocol/unlock#2356]
Optimistic or Trusted Unlocking: At the time of transaction we quickly assess whether a transaction is likely to be confirmed in a reasonable amount of time. If it is likely enough the paywall will unlock the page and confirm the transactions in the background and show the status in the Unlock flag.

It should also be noted that the likeliness of the transaction going through should quickly decline with time. The reasoning being if the transaction is not confirmed quickly the "purchaser" has more time to "double spend" the funds which would make the transaction fail altogether.

More details in this [gdoc](https://docs.google.com/document/d/1oNna06wdbcfjyNW17ghDWHTQ8vop_jYaIhw4QUFlHRo/).

## Creator Flow

- Creators should have an option of how optimistic their unlocking should be. This will factor into the algorithm.
- OFF Not Optimist: don’t unlock until confirmed
- ON let the algorithm determine the best unlocking mechanism
- I’ve also called it Trusted Unlocking in the UI
- [Mockup](https://www.figma.com/file/gmhVq6Pqeu2gH7e3FC2lr0ax/Unlock-Dashboard?node-id=0%3A1457)
![image](https://user-images.githubusercontent.com/1377389/55011565-6f398f80-4fbc-11e9-84e5-e9de00e28434.png)

## User Flow
- User comes to a locked page
- User initiates payment, metamask opens
- User sends transaction, metamask closes
- Algorithm determines if its trusted-user, unsure-user or suspicious-user 
- trusted-user: page is immediately unlocks, while showing confirmation progress in  the unlock flag
- new-user: page unlocks after a certain time/count of confirmations or page unlocks only after transaction is confirmed
- [Mockup](https://www.figma.com/proto/LkWSSnCGlKEITBB5T5NYXW/Unlock-Paywall?node-id=0%3A817&viewport=1521%2C-1188%2C0.253176&scaling=min-zoom): Click Purchase then in MetaMask Trusted Purchase
![image](https://user-images.githubusercontent.com/1377389/55012532-f20f1a00-4fbd-11e9-98ee-b7b74316d1b4.png)

### Relocking the Page
- If a trusted-users transaction is not confirmed (due to suspicious activity or confirmation taking too long) the page will lock again and the user will see informative messaging to what happened
- [Mockup](https://www.figma.com/proto/LkWSSnCGlKEITBB5T5NYXW/Unlock-Paywall?node-id=289%3A0&viewport=1521%2C-1188%2C0.253176&scaling=min-zoom): Select "Failed Transaction"
![image](https://user-images.githubusercontent.com/1377389/55012119-4d8cd800-4fbd-11e9-96b0-9457c9a991af.png)

**Comment by cellog** [comment]
this is implemented, need a new issue for mobile design
